libpod/container_internal_linux.go

```diff
@@ -174,24 +174,35 @@ func (c *Container) prepare() error {
 		return err
 	}
 
-	// Ensure container entrypoint is created (if required)
+	workdir := c.WorkingDir()
+	resolvedWorkdir, err := securejoin.SecureJoin(c.state.Mountpoint, c.WorkingDir())
```

Comment:
how would it work with volumes?

if we have something like `podman run -v /foo:/foo -w /foo/bar ...` the working dir must be looked up in the volume mount

Reply:
Very good point.

To make this work, we can reuse the copy code in `pkg/domain/infra/abi/containers_stat.go` and move it into `libpod`.  But I think that should be a separate PR.

@mheon WDYT?

Reply:
Actually, we need to tackle this now to make system tests happy.

Reply:
Would your intention be to replace the copy-up code wholesale, or just move the bit that determines where the path in a container lives?

Reply:
Just the latter.